sympy/solvers/ode/single.py

```diff
@@ -359,6 +358,143 @@ def inverse(self):
+    >>> dsolve(cos(f(x)) - (x*sin(f(x)) - f(x)**2)*f(x).diff(x),f(x), hint='1st_exact')
+    Eq(x*cos(f(x)) + f(x)**3/3, C1)
+
+    References
+    ==========
+
+    - "Differential Equations With Applications And Historical Notes",
+       Second Edition, George.F.Simmons
+    - https://en.wikipedia.org/wiki/Exact_differential_equation
+
+    # indirect doctest
+
+    """
+    hint = "1st_exact"
+    has_integral = True
+    order = [1]
+
+    def _wilds(self, f, x, order):
+        P = Wild('P', exclude=[f(x).diff(x)])
+        Q = Wild('Q', exclude=[f(x).diff(x)])
+        return P, Q
+
+    def _equation(self, fx, x, order):
+        P, Q = self.wilds()
+        return P + Q*fx.diff(x)
+
+    def _verify(self, fx) -> bool:
+        P, Q = self.wilds()
+        x = self.ode_problem.sym
+        y = Dummy('y')
+
+        m, n = self.wilds_match()
+
+        try:
+            if simplify(m) != 0:
+                m = m.subs(fx,y)
+                n = n.subs(fx,y)
+                numerator= simplify(m.diff(y)-n.diff(x))
+                # The following few conditions try to convert a non-exact
+                # differential equation into an exact one.
+                # References :
+                #1. Differential equations with applications
+                # and historical notes - George E. Simmons
+                #2. https://math.okstate.edu/people/binegar/2233-S99/2233-l12.pdf
+                if numerator:
+                    # If (dP/dy - dQ/dx) / Q = f(x)
+                    # then exp(integral(f(x))*equation becomes exact
+                    factor = simplify(numerator/n)
+                    variables = factor.free_symbols
+                    if len(variables) == 1 and x == variables.pop():
+                        factor = exp(Integral(factor).doit())
+                        m *= factor
+                        n *= factor
+                        self._wilds_match[P] = m.subs(y, fx)
+                        self._wilds_match[Q] = n.subs(y, fx)
+                        return True
+                    else:
+                        # If (dP/dy - dQ/dx) / -P = f(y)
+                        # then exp(integral(f(y))*equation becomes exact
+                        factor = simplify(-numerator / m)
+                        variables = factor.free_symbols
+                        if len(variables) == 1 and y == variables.pop():
+                            factor = exp(Integral(factor).doit())
+                            m *= factor
+                            n *= factor
+                            self._wilds_match[P] = m.subs(y, fx)
+                            self._wilds_match[Q] = n.subs(y, fx)
+                            return True
+                else:
+                    return True
+        except NotImplementedError:
+            # Differentiating the coefficients might fail because of things
+            # like f(2*x).diff(x).  See issue 4624 and issue 4719.
+            pass
```

Comment:
This try block is too broad. There is too much code in between the try and the except. We should only catch exceptions around the narrowest block of code possible (ideally a single function call). Where does the NotImplemented error come from? Maybe the try/except can just be removed.

Reply:
Are the tests failing because of my `master` branch being out of sync with sympy's `master`?